### PR TITLE
Preview package re-architecture

### PR DESCRIPTION
**Pull request by ndelangen**
move code around so packages used in the preview now have their code co-lated enough so that we can bundle them all, without duplicating code.

We need to perform more optimizations to make this work.

Storyshots seems busted, need to investigate why.

### CODE REVIEW

**Review by tmeasday — APPROVED** [review]
Awesome @ndelangen! I have a couple questions, but I think they can be addressed in upcoming PRs.

File: code/frameworks/angular/src/client/decorateStory.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { DecoratorFunction, LegacyStoryFn, StoryContext } from '@storybook/types';
-import { sanitizeStoryContextUpdate } from '@storybook/store';
+import { sanitizeStoryContextUpdate } from '@storybook/preview/dist/store';
```
[the remark this answers is not in the document]
Comment (ndelangen) [reply code/frameworks/angular/src/client/decorateStory.ts]: This has been reverted 👍

**Comment by ndelangen** [comment]
Ok, if this goes green, I have a pretty solid understanding of how it all should work.
Is does require some builder magic / changes.

I'd have to look at how to implement these changes for the vite builder.

**Comment by tmeasday** [comment]
Can you talk to the builder magic @ndelangen ?

**Comment by ndelangen** [comment]
It's green! NICE! @tmeasday yes!

**Comment by ndelangen** [comment]
replaced by https://github.com/storybookjs/storybook/pull/19718

**Pull request closed without merging**